# scNAE: Network activity evaluation reveals significant gene regulatory architectures during SARS-CoV-2 viral infec-tion from dynamic scRNA-seq data #


**scNAE will be detailed if the manuscript is accepted!**

**Please write to [zpliu@sdu.edu.cn](mailto:zpliu@sdu.edu.cn) if you have any questions.**

**Please cite our paper if it helps you.**

![workfolw](https://github.com/zpliulab/scNAE/blob/main/Supplementary/single%20cell.jpg)

Computer Environment [![MATLAB](https://img.shields.io/badge/MATLAB-R2022a-green.svg "MATLAB")](https://ww2.mathworks.cn/products/matlab.html "MATLAB"):
-
- MATLAB R2022a (requires Parallel Computing Toolbox installed)
- If your working environment cannot enable parallel pooling, please replace *parfor* with *for*


File:
-
- code: subfunction used in the scNAE process.
- result: save results.
- simu_data: simulation datasets.
- Supplementary
- demo.m: run the simulation example

The core interface function of scNAE is code\refer_scNAE.m

    [newgra,obj,opt] = refer_scNAE(data,network,Lp,parameter);

where *data.orig* represents the gene expression profile of the time series; *data.fit* represents the corresponding differential value;*network* is the network to be evaluated; *Lp* represents the norm type used; *parameter* is a structure, where parameter.i is the regularization parameter that must be set. The return value *newgra* is the inference network, *obj* is the loss value generated by each iteration, and *opt* is the parameter used.

