# scNAE: Network activity evaluation reveals significant gene regulatory architectures during SARS-CoV-2 viral infec-tion from dynamic scRNA-seq data #


**scNAE will be detailed if the manuscript is accepted!**

**Please write to [zpliu@sdu.edu.cn](mailto:zpliu@sdu.edu.cn) if you have any questions.**

**Please cite our paper if it helps you.**

![workfolw](https://github.com/zpliulab/scNAE/blob/main/Supplementary/single%20cell.jpg)

Computer Environment [![MATLAB](https://img.shields.io/badge/MATLAB-R2022a-green.svg "MATLAB")](https://ww2.mathworks.cn/products/matlab.html "MATLAB"):
-
- MATLAB R2022a (requires Parallel Computing Toolbox installed)
- If your working environment cannot enable parallel pooling, please replace *parfor* with *for*


## Files:
- **code: subfunction used in the scNAE process.**
- **result: save results.**
- **simu_data: simulation datasets.**
- **Supplementary**
- **demo.m: run the simulation example**
- 
## Run the code
Run the following code in the R environment
- demo.m (Note that "simu_data_302.mat" in the simulated data is assumed to have 30 genes and 2 cell clusters. Other data are named similarly.)
- demo_SARS-CoV-2.m
- 
## Use scNAE on a new dataset
The core interface function of scNAE is code\refer_scNAE.m
```[newgra,obj,opt] = refer_scNAE(data,network,Lp,parameter);```

	Input:
	  data : A structure variable containing the original value and the differential value, *data.orig* represents the gene expression profile of the time series; *data.fit* represents the corresponding differential value;
	  network:   An (m*m) adjacency matrix
	  Lp:   the Lp-norm type used, p can be selected {0, 0~1, 2}
      parameterï¼š *parameter* is a structure, where 'parameter.i' is the regularization parameter that must be set., e.g: parameter.i=1e-6 ~ 1e2
      
	Output:
	 newgra: Reconstructed adjacency matrix.
     obj: The loss value generated by each iteration (used to calculate p-value)
     opt: optimal parameters

